controllers/topicPage/TopicPage.js

```diff
@@ -19,17 +19,25 @@ class TopicPage {
   }
 
   async getFeedTopicPages(payload) {
+    let results = {
+      noActivity: false,
+      data: []
+    };
     const topicPages = await this._getStreamService.getTopicPages(
       payload.id,
       payload.limit,
       payload.offset
     );
+    if (!topicPages.length) {
+      results.noActivity = true;
+      return results;
+    }
     let newTopicPagesWithBlock = new BlockServices(
       payload.listBlockUser,
       payload.listBlockDomain,
       payload.listPostAnonymous
     ).getHasBlock(topicPages);
-    let results = await filterFeeds(payload.userId, newTopicPagesWithBlock, payload.id);
+    results.data = await filterFeeds(payload.userId, newTopicPagesWithBlock, payload.id);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a check for empty `topicPages` and sets the `noActivity` flag accordingly. This is a good improvement as it handles the case where there are no topic pages. However, error handling seems to be missing in case the `getTopicPages` function fails. It would be better to wrap this call in a try-catch block to handle any potential errors.

```diff
     const topicPages = await this._getStreamService.getTopicPages(
       payload.id,
       payload.limit,
       payload.offset
     );
+     try {
+       const topicPages = await this._getStreamService.getTopicPages(
+         payload.id,
+         payload.limit,
+         payload.offset
+       );
+     } catch (error) {
+       console.error('Error fetching topic pages:', error);
+       results.noActivity = true;
+       return results;
+     }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->